### ISSUE TITLE
Python client improvements

### ISSUE THREAD

**Issue by BrunoBonacci** [samsara/samsara#63]
Here some improvements:

  - [ ] fix python import (`samsara_sdk.client`) in `docs/clients/python-client.md`
  - [ ] add some doc in `README.md` on how to build locally
  - [ ] add some doc in `README.md` on how to publish the library to the world
  - [ ] update `setup.py/get_long_description` to point to `docs/clients/python-client.md`
  - [ ] bind version to `../../samsara.version`
  - [ ] `_flush_buffer` is locking for the duration of the POST (maybe a few seconds) https://github.com/samsara/samsara/blob/master/clients/python/samsara_sdk/client.py#L193, replace this with a much narrower lock as suggested in PR.
```
        with self.lock:
            events = self._buffer
            last_id, count = self._buffer.last_event_id, len(events)

        success = self.publish_events(events)
```
  - [ ] `del self[0]['__id__']` I suspect this doesn't do much. (https://github.com/samsara/samsara/blob/master/clients/python/samsara_sdk/helpers.py#L135), I think this should be moved to before the call `self.publish_events(self._buffer)` in https://github.com/samsara/samsara/blob/master/clients/python/samsara_sdk/client.py#L194
  - [ ] add in the doc an example of how to get the current time in millis, maybe an helper function could be nice `current_time_millis`

**Comment by puhrez** [comment]
re your second to last bullet, it's actually the python way for `dissoc` 
```
In [2]: d = {'__id__': 1, 'eventName': 'user.did.something'}

In [3]: d
Out[3]: {'__id__': 1, 'eventName': 'user.did.something'}

In [4]: del d['__id__']

In [5]: d
Out[5]: {'eventName': 'user.did.something'}

```

**Comment by puhrez** [comment]
Is point 3 appropriate for a package's readme, as in I haven't found any library with the steps to publish the library to a certain package repo.

**Comment by BrunoBonacci** [comment]
I understand what `del` does, however I think that where it is places it hasn't the effect I was asking for.
 from the code it looks like:

  - you increment a counter and add events in the buffer with the counter as `__id__`
  - in `flush_buffer` you read the content of the buffer (all events have `__id__`)
  - you POST the events with `pusblish_events` (all events still have `__id__`)
  - then you remove the events which have been posted and delete the `__id__`

so the attribute `__id__` it seems it get published as well which I wouldn't like it as it is meaningless.
The right point for the `del` of the `__id__` is in `flush_buffer` between the 2 and 3 step.

**Comment by BrunoBonacci** [comment]
for point 2 and 3, I will be testing/releasing the library, and I have no idea how to do it.
If you think README isn't a good place for it, please add another file like `/clients/python/doc/RELEASE.md`

**Comment by puhrez** [comment]
this is what i've used to test the package submission : http://peterdowns.com/posts/first-time-with-pypi.html

**Comment by BrunoBonacci** [comment]
merged #64
